src/csexp_rpc/csexp_rpc.mli

```diff
@@ -15,6 +15,22 @@
 
 open Stdune
 
+module type Worker = sig
```

Comment:
Sorry, I don't understand. What is a "worker"? Could you add some comments?

Reply:
My guess is it's a way to inject arbitrary functions into Dune's scheduler but I'm not quite sure what guarantees this interface is supposed to provide.

Reply:
Btw, does it make sense to make `Csexp_rpc` abstract over the underlying concurrency monad, i.e. turn it into a functor that accepts the monad and maybe some bits of the scheduler? Then, I think the API might look a bit more natural.

Reply:
"Worker" is used to run jobs in background threads and report the results in the main thread. In this case, it's needed to implement non-blocking IO in a portable way.

> Btw, does it make sense to make Csexp_rpc abstract over the underlying concurrency monad, i.e. turn it into a functor that accepts a monad and maybe some bits of the scheduler? Then, I think the API might look a bit more natural.

I'd rather only do this if there's a concrete benefit to somebody from this. In other words, someone wants to use this library without Fiber. Given that it's trivial enough to implement in anyone's favorite variant of concurrency library, it seems like the use case is low.

>  Then, I think the API might look a bit more natural.

Perhaps it's natural because we're using threads for background IO? If so I agree, but I think the solution would be to add proper IO primitives to the Scheduler them.

Reply:
Ack, thanks.